### PR TITLE
Adding wildcard policy to `oc create route` and a column for it in `oc get route`

### PR DESCRIPTION
**Pull request by JacobTanenbaum**
Adding wildcard policy to 'oc create route'

Adding a column for wildcard policy in 'oc get route'

Bug 1391786 [Link](https://bugzilla.redhat.com/show_bug.cgi?id=1391786)

### CODE REVIEW

**Comment by JacobTanenbaum** [comment]
@openshift/networking

**Comment by knobunc** [comment]
[test]

**Comment by ncdc** [comment]
Please reword your commit message to be descriptive as to what it's actually doing (Add wildcard policy to `oc create route`), and do the same for the PR title as well. Thanks!

File: pkg/cmd/cli/cmd/create/route.go [hunk 1]
```diff
@@ -409,3 +430,10 @@ func resolveRouteName(args []string) (string, error) {
 	}
 	return "", nil
 }
+
+func resolveWildcardPolicy(wildcardpolicy string) (string, error) {
+	if len(wildcardpolicy) > 0 && (wildcardpolicy != "Subdomain" && wildcardpolicy != "None") {
+		return "", fmt.Errorf("only \"Subdomain\" or \"None\" are supported for wildcardpolicy")
+	}
+	return wildcardpolicy, nil
+}
```
Comment (ramr) [inline pkg/cmd/cli/cmd/create/route.go]: Maybe this should use validation from the route api (we will need to `export` something out in the route validation for that though). This looks good for now but if you can please add a comment here that we should use the validation from the route.  Thx

Comment (JacobTanenbaum) [inline pkg/cmd/cli/cmd/create/route.go]: The validation functions are called as part of object creation on the server side. I used debug functions to print a stack trace on the server side on a failed attempt to create a route (for another PR but should be the same for this one) [StackTrace](http://pastebin.com/YcRLdKyA)

**Comment by knobunc** [comment]
Ok, we will open a new bug because 'oc route' doesn't validate anything on the cli side.  This PR is consistent with everything else in the 'oc route' so I think it is good to go.  When this is squashed and the description cleaned up, I think it is good to go.  Thanks.

**Comment by JacobTanenbaum** [comment]
The BZ that ben mentioned above [BZ1418023](https://bugzilla.redhat.com/show_bug.cgi?id=1418023)

**Comment by ramr** [comment]
@knobunc adding some of these checks on client side could be problematic. One is that older cli clients would behave differently vis-a-vis newer cli clients + what happens when the server validation changes say for a list of choices from [foo, bar] to [foo, bar, baz] or even [one, two]? The older cli client checks would be invalid at that point or worse yet even block valid values.

**Comment by knobunc** [comment]
@ramr: All good points.  Added your concerns to the bug.

[merge]

File: pkg/cmd/cli/cmd/create/route.go [hunk 2]
```diff
@@ -112,6 +113,11 @@ func CreateEdgeRoute(f *clientcmd.Factory, out io.Writer, cmd *cobra.Command, ar
 		return err
 	}
 
+	wildcardpolicy := kcmdutil.GetFlagString(cmd, "wildcardpolicy")
```
Comment (smarterclayton) [inline pkg/cmd/cli/cmd/create/route.go]: This should be wildcard-policy.  Please open a follow-up.

File: pkg/cmd/cli/describe/printer.go [hunk 3]
```diff
@@ -36,7 +36,7 @@ var (
 	imageStreamImageColumns = []string{"NAME", "DOCKER REF", "UPDATED", "IMAGENAME"}
 	imageStreamColumns      = []string{"NAME", "DOCKER REPO", "TAGS", "UPDATED"}
 	projectColumns          = []string{"NAME", "DISPLAY NAME", "STATUS"}
-	routeColumns            = []string{"NAME", "HOST/PORT", "PATH", "SERVICES", "PORT", "TERMINATION"}
+	routeColumns            = []string{"NAME", "HOST/PORT", "PATH", "SERVICES", "PORT", "TERMINATION", "WILDCARDPOLICY"}
```
Comment (smarterclayton) [inline pkg/cmd/cli/describe/printer.go]: WILDCARD, column length is critical and the extra POLICY doesn't help much.

**Comment by smarterclayton** [comment]
Please ensure @openshift/cli-review is included on all CLI changes to enforce consistency for our processes.

**Comment by JacobTanenbaum** [comment]
@smarterclayton posted follow up: https://github.com/openshift/origin/pull/12774

**Pull request merged**